### PR TITLE
Add 'configure_permissions_ownership' function to 'libfs.sh'

### PR DESCRIPTION
**Pull request by juan131**
This PR adds a new function `configure_permissions_ownership` to the library **'libos.sh'**:

This new function allows to configure the permissions/ownership of a set of paths.

### CODE REVIEW

**Comment by alemorcuq** [comment]
It would be nice if it was possible to change the permissions and not the owners.

File: stretch/rootfs/libfs.sh [hunk 1]
```diff
@@ -53,3 +53,80 @@ is_dir_empty() {
...
+    local -r dir_mode="${2:-777}"
+    local -r file_mode="${3:-666}"
+
+    read -r -a filepaths <<< "$paths"
+    for p in "${filepaths[@]}"; do
+        if [[ -e "$p" ]]; then
+            find -L "$p" -type d -exec chmod "$dir_mode" {} \;
+            find -L "$p" -type f -exec chmod "$file_mode" {} \;
+        else
+            warn "$p do not exist!!"
```
Comment (alemorcuq) [inline stretch/rootfs/libfs.sh]: `warn "$p does not exist!!"`

File: stretch/rootfs/libfs.sh [hunk 2]
```diff
@@ -53,3 +53,80 @@ is_dir_empty() {
...
+    local -r paths="${1:?paths is missing}"
+    local -r user="${2:-$(id -u)}"
+    local -r group="${3:-$(id -g)}"
+
+    read -r -a filepaths <<< "$paths"
+    for p in "${filepaths[@]}"; do
+      if [[ -e "$p" ]]; then
+          chown -LR "$user":"$group" "$p"
+      else
+          warn "$p do not exist!!"
```
Comment (alemorcuq) [inline stretch/rootfs/libfs.sh]: `warn "$p does not exist!!"`

File: stretch/rootfs/libfs.sh [hunk 3]
```diff
@@ -53,3 +56,66 @@ is_dir_empty() {
...
+    read -r -a filepaths <<< "$paths"
+    for p in "${filepaths[@]}"; do
+        if [[ -e "$p" ]]; then
+            [[ -n $dir_mode ]] && find -L "$p" -type d -exec chmod "$dir_mode" {} \;
+            [[ -n $file_mode ]] && find -L "$p" -type f -exec chmod "$file_mode" {} \;
+            [[ -n $user ]] && [[ -n $group ]] && chown -LR "$user":"$group" "$p"
+            [[ -n $user ]] && [[ -z $group ]] && chown -LR "$user" "$p"
+            [[ -z $user ]] && [[ -n $group ]] && chgrp -LR "$group" "$p"
+        else
+            warn "$p does not exist!!"
```
Comment (marcosbc) [inline stretch/rootfs/libfs.sh]:
```suggestion
            warn "$p does not exist"
```

File: stretch/rootfs/libfs.sh [hunk 4]
```diff
@@ -53,3 +56,66 @@ is_dir_empty() {
...
+    shift 1
+    while [ "$#" -gt 0 ]; do
+        case "$1" in
+            -f|--file-mode)
+                shift
+                file_mode="${1:?missing mode for directories}"
+                ;;
+            -d|--dir-mode)
+                shift
+                dir_mode="${1:?missing mode for files}"
```
Comment (marcosbc) [inline stretch/rootfs/libfs.sh]:
```suggestion
                dir_mode="${1:?missing mode for directories}"
```

File: stretch/rootfs/libfs.sh [hunk 5]
```diff
@@ -53,3 +56,66 @@ is_dir_empty() {
...
+    local user=""
+    local group=""
+
+    # Validate arguments
+    shift 1
+    while [ "$#" -gt 0 ]; do
+        case "$1" in
+            -f|--file-mode)
+                shift
+                file_mode="${1:?missing mode for directories}"
```
Comment (marcosbc) [inline stretch/rootfs/libfs.sh]:
```suggestion
                file_mode="${1:?missing mode for files}"
```

File: stretch/rootfs/libfs.sh [hunk 6]
```diff
@@ -53,3 +56,66 @@ is_dir_empty() {
...
+                ;;
+            *)
+                echo "Invalid command line flag $1" >&2
+                return 1
+                ;;
+        esac
+        shift
+    done
+
+    read -r -a filepaths <<< "$paths"
```
Comment (marcosbc) [inline stretch/rootfs/libfs.sh]: Why can't we just run `for p in "${paths[$@]}"`?

Comment (juan131) [reply stretch/rootfs/libfs.sh]: Nope. It's treated as a string before this conversion.

**Comment by juan131** [comment]
> we usually want to configure permissions in all cases, but only changing the ownership when running as root user

The you can simply choose to indicate the `-d` and `-f` flags omitting `-u` and `-g`

**Pull request merged**